### PR TITLE
chaosctl add physical-machine command

### PR DESCRIPTION
**Pull request by iguoyr**
Signed-off-by: SiyuChen <ryougi201@gmail.com>

### What problem does this PR solve?

as #2390 said, this pr prepares for physical machine auth.

### What's changed and how it works?

Proposal: [physical-machine-auth](https://github.com/chaos-mesh/rfcs/blob/be5f6566bde0fe4d709173f85a6cc2e4848f91c3/text/2021-09-09-physical-machine-auth.md)

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by iguoyr** [comment]
/cc @WangXiangUSTC @STRRL 
PTAL

File: pkg/chaosctl/physicalmachine/ssh.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		port:   port,
+	}, nil
+}
+
+func (s *SshTunnel) Open() error {
+	// first, try ssh with public keys
+	conn, err := ssh.Dial("tcp", net.JoinHostPort(s.host, s.port), s.config)
+	if err != nil {
+		// if failed, try ssh with password
+		fmt.Printf("please input the password: ")
```
Comment (YangKeao) [inline pkg/chaosctl/physicalmachine/ssh.go:76]: Maybe we should use `"golang.org/x/term".ReadPassword`?

**Comment by YangKeao** [comment]
Rest LGTM

File: pkg/chaosctl/physicalmachine/ssh.go (hunk 1 above)
Comment (YangKeao) [reply pkg/chaosctl/physicalmachine/ssh.go:76]: BTW, why don't we add multiple `AuthMethod` in the list in order? They will be tried one by one if I'm not wrong.

The `Password` authantication can be set to the `PasswordCallback`, and use "golang.org/x/term".ReadPassword to get the password.

**Review by YangKeao — COMMENTED** [review]
Rest LGTM

File: pkg/chaosctl/physicalmachine/generate.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := generateOption.Validate(); err != nil {
+				return err
+			}
+			return generateOption.Run()
+		},
+	}
+	generateCmd.PersistentFlags().StringVar(&generateOption.outputPath, "path", "/etc/chaosd/pki", "path to save generated certs")
+	generateCmd.PersistentFlags().StringVar(&generateOption.caCertFile, "cacert", "", "file path to cacert file")
+	generateCmd.PersistentFlags().StringVar(&generateOption.caKeyFile, "cakey", "", "file path to cakey file")
```
Comment (WangXiangUSTC) [inline pkg/chaosctl/physicalmachine/generate.go:55]: Should be the `file name`? The path is already set above.

Comment (WangXiangUSTC) [reply pkg/chaosctl/physicalmachine/generate.go:55]: and maybe it's better to add a default value.

File: pkg/chaosctl/physicalmachine/ssh.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		host:   ip,
+		port:   port,
+	}, nil
+}
+
+func (s *SshTunnel) Open() error {
+	// first, try ssh with public keys
+	conn, err := ssh.Dial("tcp", net.JoinHostPort(s.host, s.port), s.config)
+	if err != nil {
+		// if failed, try ssh with password
```
Comment (WangXiangUSTC) [inline pkg/chaosctl/physicalmachine/ssh.go:75]: how about printing the error message?

**Review by WangXiangUSTC — COMMENTED** [review]
Could you please add an example in the description to show when using this command? 

rest LGTM

File: pkg/chaosctl/physicalmachine/generate.go (hunk 2 above)
Comment (iguoyr) [reply pkg/chaosctl/physicalmachine/generate.go:55]: `--path` is the path for output file (e.g. chaosd.crt and chaosd.key), but caCert and caKey are the input file in the command.

Comment (iguoyr) [reply pkg/chaosctl/physicalmachine/generate.go:55]: > Could you please add an example in the description to show when using this command?

I'm working on the document about this feature!

File: pkg/chaosctl/physicalmachine/ssh.go (hunk 3 above)
Comment (iguoyr) [reply pkg/chaosctl/physicalmachine/ssh.go]: As https://github.com/chaos-mesh/chaos-mesh/pull/2642#discussion_r775411415 suggested, I have already removed this retry logic :)

File: pkg/chaosctl/physicalmachine/ssh.go (hunk 1 above)
Comment (iguoyr) [reply pkg/chaosctl/physicalmachine/ssh.go]: fixed

File: pkg/chaosctl/physicalmachine/ssh.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+	}
+	defer f.Close()
+
+	if _, err := f.Write(data); err != nil {
+		return errors.Wrapf(err, "write file %s failed", filename)
+	}
+	return nil
+}
+
+func getHostKey(host string) (ssh.PublicKey, error) {
```
Comment (YangKeao) [inline pkg/chaosctl/physicalmachine/ssh.go:125]: This function can be replaced by `"golang.org/x/crypto/knownhosts".New()`.

**Review by YangKeao — COMMENTED** [review]
Github VSCode plugin eats some of my comments last time😢 

Rest LGTM.

File: pkg/chaosctl/physicalmachine/ssh.go (hunk 4 above)
Comment (iguoyr) [reply pkg/chaosctl/physicalmachine/ssh.go]: fixed!

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Pull request merged**